webpack/react_app/components/TargetingHosts/index.js

```diff
@@ -1,37 +1,99 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
+
+import SearchBar from 'foremanReact/components/SearchBar';
+import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
+import { useForemanSettings } from 'foremanReact/Root/Context/ForemanContext';
 import { stopInterval } from 'foremanReact/redux/middlewares/IntervalMiddleware';
+
 import TargetingHosts from './TargetingHosts';
 
 import {
   selectItems,
   selectStatus,
   selectAutoRefresh,
+  selectTotalHosts,
 } from './TargetingHostsSelectors';
+import { getApiUrl } from './TargetingHostsHelpers';
 import { getData } from './TargetingHostsActions';
 import { TARGETING_HOSTS } from './TargetingHostsConsts';
 
 const WrappedTargetingHosts = () => {
   const dispatch = useDispatch();
+  const { perPage, perPageOptions } = useForemanSettings();
+
   const autoRefresh = useSelector(selectAutoRefresh);
   const items = useSelector(selectItems);
   const status = useSelector(selectStatus);
+  const totalHosts = useSelector(selectTotalHosts);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [pagination, setPagination] = useState({
+    page: 1,
+    perPage,
+    perPageOptions,
+  });
+  const [apiUrl, setApiUrl] = useState(getApiUrl(searchQuery, pagination));
+
+  const handleSearch = query => {
+    const defaultPagination = { page: 1, perPage: pagination.perPage };
+    dispatch(stopInterval(TARGETING_HOSTS));
+
+    setApiUrl(getApiUrl(query, defaultPagination));
+    setSearchQuery(query);
+    setPagination(defaultPagination);
+  };
+
+  const handlePagination = args => {
+    dispatch(stopInterval(TARGETING_HOSTS));
```

Comment:
Should only run is the interval exists otherwise it throws an error that the interval doesnt exists 
You can select it using `foremanReact/redux/middlewares/IntervalMiddleware/IntervalSelectors.js`

Reply:
I wasn't able reproduce the errors, but being 100% sure added new function
```
+  const stopApiInterval = () => {
+    if (intervalExists) {
+      dispatch(stopInterval(TARGETING_HOSTS));
+    }
+  };

```